cleanlab/rank.py

```diff
@@ -572,21 +572,27 @@ def get_knn_distance_ood_scores(
     nbrs : sklearn.neighbors.NearestNeighbors
       Instantiated NearestNeighbors class object that's been fitted on a dataset in the same feature space.
       Note that the distance metric and n_neighbors is specified when instantiating this class.
+      If nbrs=None, then by default nbrs=sklearn.neighbors.NearestNeighbors(n_neighbors=k, metric="cosine").fit(features)
       See: https://scikit-learn.org/stable/modules/neighbors.html
 
     k : int, default=None
       Number of neighbors to use when calculating average distance to neighbors.
       This value k needs to be less than or equal to max_k which is the n_neighbors used when fitting instantiated NearestNeighbors class object.
       If k=None, then by default k=min(10, max_k) is used where max_k is extracted from the given nbrs.
+      If nbrs is not provided, then by default k=10.
 
     Returns
     -------
     avg_nbrs_distances : np.array
       Average distance to k-nearest neighbors for each datapoint which is used as a score for OOD detection.
     """
+    # if nbrs is not provided, then use default KNN classifier
+    if nbrs is None:
+        n_neighbors = k if k is not None else 10
+        nbrs = NearestNeighbors(n_neighbors=n_neighbors, metric="cosine").fit(features)
 
     # number of neighbors specified when fitting instantiated NearestNeighbors class object
-    max_k = nbrs.n_neighbors
+    max_k = nbrs.n_neighbors if nbrs else 10
```

Comment:
I think max_k should also take into account if k was specified or not. 
I.e. 
```
if nbrs:
    max_k = nbrs.n_neighbors
elif k is not None:
    max_k = k
else:
    max_k = 10
```

Reply:
Ah scratch that ^ if the code is already implementing the same logic but in a different way?

I guess the odd part in the way the code is structured right now is why would the `else 10` clause ever be triggered?

Based on the existing code (copied below) it seems that cannot happen or maybe I'm just missing something:

```
if nbrs is None:
        n_neighbors = k if k is not None else 10
        nbrs = NearestNeighbors(n_neighbors=n_neighbors, metric="cosine").fit(features)

    # number of neighbors specified when fitting instantiated NearestNeighbors class object
    max_k = nbrs.n_neighbors if nbrs else 10
```

Reply:
+1. Cleaner construction might be to just set `k=10` in the function definition.

Then:
```
if nbrs is None:
  nbrs = NearestNeighbors(n_neighbors=k, metric="cosine").fit(features)
```

and
```
if k > nbrs.n_neighbors:
  k = nbrs.n_neighbors
  [warn that we cap k at nbrs.n_neighbors]

avg_nbrs_distances = distances[:, :k].mean(axis=1)

```

Reply:
Way cleaner thank you guys!